src/core/qgsdistancearea.cpp

```diff
@@ -537,6 +540,111 @@ double QgsDistanceArea::measureLine( const QgsPoint &p1, const QgsPoint &p2, Qgs
   return result;
 }
 
+double QgsDistanceArea::measureLineProjected( const QgsPoint &p1, double distance, double azimuth, QgsPoint *projected_point )
+{
+  double result = 0.0;
+  QgsPoint p2;
+  if ( geographic() )
```

Comment:
I don't like this method being non-const and silently altering the QgsDistanceArea object. This should be changed to a const method, which goes something like:

    if ( geographic() && mEllipsoidalMode && mEllipsoid != GEO_NONE )
         ... use computeSpheroidProject method
    else
         .... use cartesian method

If you're concerned about people using this with non-ellipsoidal QgsDistanceArea then just note it in the API docs and make the caller responsible for correctly creating an ellipsoidal QgsDistanceArea.